src/tty/windows.rs

```diff
@@ -276,8 +277,9 @@ impl<'a> EventedReadWrite for Pty<'a, NamedPipe, NamedPipe> {
 
 impl<'a> OnResize for Winpty<'a> {
     fn on_resize(&mut self, sizeinfo: &SizeInfo) {
-        if sizeinfo.cols().0 > 0 && sizeinfo.lines().0 > 0 {
-            self.set_size(sizeinfo.cols().0, sizeinfo.lines().0)
+        let (cols, lines) = (sizeinfo.cols().0, sizeinfo.lines().0);
+        if cols > 0 && cols < u16::MAX as usize && lines > 0 && lines < u16::MAX as usize {
```

Comment:
This should be changed to `cols <= u16::MAX` and `lines <= u16::MAX`. Doesn't really make much of a difference, but makes sense for correctness.